arango/collection.py

```diff
@@ -893,26 +942,53 @@ def find_in_box(
         assert isinstance(longitude2, Number), "longitude2 must be a number"
         assert is_none_or_int(skip), "skip must be a non-negative int"
         assert is_none_or_int(limit), "limit must be a non-negative int"
+        assert is_none_or_str(index), "index must be a string"
 
-        data: Json = {
-            "collection": self._name,
-            "latitude1": latitude1,
-            "longitude1": longitude1,
-            "latitude2": latitude2,
-            "longitude2": longitude2,
-        }
-        if skip is not None:
-            data["skip"] = skip
-        if limit is not None:
-            data["limit"] = limit
+        # TODO: REVISIT `geo_str` logic
+        # https://www.arangodb.com/docs/stable/aql/functions-geo.html#within_rectangle
+        # Initial assumption that attribute names are "latitude" and "longitude"
+        geo_str = "[doc.longitude, doc.latitude]"
         if index is not None:
-            data["geo"] = self._name + "/" + index
+            geo_index = self.get_index(f"{self.name}/{index}")
```

Comment:
I am in need of a second opinion on the new implementation of `collection.find_in_box`

Given the deprecation of the `WITHIN_RECTANGLE` AQL function, we are now opting to use the `GEO_CONTAINS` method in combination with the `GEO_POLYGON` method: https://www.arangodb.com/docs/stable/aql/functions-geo.html#within_rectangle

The `GEO_CONTAINS` method takes two parameters: the polygon, and the coordinates of the document.

Given that the attribute names representing the coordinates of the document  can vary on a user-basis, we cannot assume that `doc.longitude` and `doc.latitude` will always exist.

In order to prevent any breaking changes to `collection.find_in_box`, we rely on the existing `index` parameter to fetch the Geo Index's `fields` values to use as the document's coordinate attributes.

Therefore, we assume that the Geo Index's `fields` is an array containing either 2 values at most. Otherwise, a `ValueError` is raised.

Perhaps I am overcomplicating things; happy to move to another solution if we see one

Reply:
I agree!
Trying to create a GEO index with more than 2 attributes will throw an exception. IMO, double checking it here (along with the fact that the index is truly "geo") is a good idea.